src/helpers.rs

```diff
@@ -480,31 +480,47 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         self.eval_context_mut().memory.write_bytes(scalar, bytes.iter().copied().chain(iter::once(0u8)))?;
         Ok(true)
     }
+
+    fn alloc_os_str_as_c_str(
+        &mut self,
+        os_str: &OsStr,
+        memkind: MemoryKind<MiriMemoryKind>
+    ) -> Pointer<Tag> {
+        let bytes = os_str_to_bytes(os_str).unwrap();
+        let size = bytes.len() as u64 + 1; // Make space for `0` terminator.
+
+        let this = self.eval_context_mut();
+
+        let arg_type = this.tcx.mk_array(this.tcx.types.u8, size);
+        let arg_place = this.allocate(this.layout_of(arg_type).unwrap(), memkind);
+        self.write_os_str_to_c_str(os_str, arg_place.ptr, size).unwrap();
+        arg_place.ptr.assert_ptr()
+    }
 }
 
 #[cfg(target_os = "unix")]
-fn os_str_to_bytes<'tcx, 'a>(os_str: &'a OsStr) -> InterpResult<'tcx, &'a [u8]> {
+pub(crate) fn os_str_to_bytes<'tcx, 'a>(os_str: &'a OsStr) -> InterpResult<'tcx, &'a [u8]> {
```

Comment:
These don't have to be public now, do they?

Reply:
You are right, I will remove the `pub(crate)` annotations.